### PR TITLE
Feature/queue split

### PR DESCRIPTION
**Pull request by rfleming71**
Splitting the queue management code into the JungleQueue class. This gets the code two things:
Support for using just the queue without Publish Support
Support full queue configuration by creating the queue later.

Two things still need to be added back in after this pull request to make the changes fully backwards compatible:
MessageSerializer was removed for simplicity, the code is still there, just not used.
I updated the SNS subscriptions to use Raw Message delivery, meaning they will no longer get the SNS message envelope. Need to add a bit of conversion code to the message parser to support the SNS envelope that may already be in queue.

### CODE REVIEW

File: ConsoleTestApp/Program.cs [hunk 1]
```diff
@@ -54,21 +54,21 @@ static void Main(string[] args)
 
         static IRunJungleBus CreateRecieveOnlyBus()
```
Comment (neilfitzgerald) [inline ConsoleTestApp/Program.cs]: CreateReceiveOnlyBus

Do a search and replace for recieve and replace with receive.

File: JungleBus.Tests/Messaging/MessageProcessorTests.cs [hunk 2]
```diff
@@ -202,6 +265,22 @@ public void Handle(TransportMessage message, Exception ex)
             {
                 ++_transportMessageFaultHandler;
             }
+        }
+
+        class TestStatHandler1 : IWantMessageStatistics
+        {
+            public void RecieveStatisitics(IMessageStatistics statistics)
```
Comment (neilfitzgerald) [inline JungleBus.Tests/Messaging/MessageProcessorTests.cs]: rename to ReceiveStatistics

File: JungleBus.Tests/Messaging/MessageProcessorTests.cs [hunk 3]
```diff
@@ -202,6 +265,22 @@ public void Handle(TransportMessage message, Exception ex)
...
+        {
+            public void RecieveStatisitics(IMessageStatistics statistics)
+            {
+                ++_testStatHandler1Called;
+            }
+        }
+
+        class TestStatHandler2 : IWantMessageStatistics
+        {
+            public void RecieveStatisitics(IMessageStatistics statistics)
```
Comment (neilfitzgerald) [inline JungleBus.Tests/Messaging/MessageProcessorTests.cs]: rename to ReceiveStatistics

File: JungleBus/Aws/Sqs/SqsQueue.cs [hunk 4]
```diff
@@ -87,7 +85,7 @@ public SqsQueue(RegionEndpoint endpoint, string queueName, int retryCount, IMess
                 _simpleQueueService.SetQueueAttributes(createResponse.QueueUrl, new Dictionary<string, string>() { { "MessageRetentionPeriod", "1209600" } });
             }
 
-            MessageParser = messageParser;
+            MessageParser = new MessageParser();
             MaxNumberOfMessages = 10;
```
Comment (neilfitzgerald) [inline JungleBus/Aws/Sqs/SqsQueue.cs]: Make a parameter with a default value? Don't hard code this, push this out to config?

Comment (rfleming71) [inline JungleBus/Aws/Sqs/SqsQueue.cs]: Looking at this, pretty sure this is a bug and should be hard coded to 1. If the SQS returns more than 1 message, all messages would have the same timeout span. I think this is from when I have multithreading implemented via a single message pump. I will put in a bug for this.

**Pull request merged**